verifiers/internal/gha/npm.go

```diff
@@ -66,25 +71,62 @@ func (n *Npm) ProvenanceLeafCertificate() *x509.Certificate {
 	return n.verifiedProvenanceAtt.SigningCert
 }
 
-func NpmNew(ctx context.Context, root *TrustedRoot, attestationBytes []byte) (*Npm, error) {
+// NpmNew creates a new Npm verifier.
+func NpmNew(ctx context.Context, root *TrustedRoot, attestationBytes []byte, verifierOptioners ...options.VerifierOptioner) (*Npm, error) {
 	var aSet attestationSet
 	if err := json.Unmarshal(attestationBytes, &aSet); err != nil {
 		return nil, fmt.Errorf("%w: json.Unmarshal: %v", errrorInvalidAttestations, err)
 	}
-
 	prov, pub, err := extractAttestations(aSet.Attestations)
 	if err != nil {
 		return nil, err
 	}
+	verifierOpts, err := getVerifierOpts(verifierOptioners...)
+	if err != nil {
+		return nil, err
+	}
 	return &Npm{
 		ctx:  ctx,
 		root: root,
 
 		provenanceAttestation: prov,
 		publishAttestation:    pub,
+		verifierOpts:          verifierOpts,
 	}, nil
 }
 
+// getDefaultSigstoreTUFClient returns the default Sigstore TUF client.
```

Comment:
It might be good to export a function that gives the default options so its more explicit to an interface user what's going on? Like what TUF does: https://github.com/sigstore/sigstore-go/blob/v0.4.0/pkg/tuf/options.go#L121.

Reply:
Added